src/nat/runtime/metrics.py

```
@@ -0,0 +1,199 @@
+
+        Returns:
+            PerUserMonitorResponse with all user metrics
+        """
+        users: list[PerUserResourceUsage] = []
+
+        async with self._session_manager._per_user_builders_lock:
+            for user_id, builder_info in self._session_manager._per_user_builders.items():
+                try:
+                    user_metrics = self._build_user_metrics(user_id, builder_info)
+                    users.append(user_metrics)
+                except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)